sdk/python/lib/pulumi/output.py

```diff
@@ -224,9 +224,12 @@ async def run() -> U:
                     # Try and set the result. This might fail if we're shutting down,
```

Comment:
I don't understand the "This might fail if we're shutting down" comment, perhaps we can try dropping the try..except now that we have extra .done() guards.

Reply:
L202 await on _is_known was causing an exception after result_resources was already set; result_resources.set_result(resources) threw another exception short-circuiting backup setters for is_known and Is_secret.